src/arithmetic.jl

```diff
@@ -33,20 +33,25 @@ for T in (:Taylor1, :HomogeneousPolynomial, :TaylorN)
 end
 
 ## zero and one ##
-for T in (:Taylor1, :TaylorN), f in (:zero, :one)
-    @eval ($f)(a::$T) = $T(($f)(a[0]), a.order)
+for T in (:Taylor1, :TaylorN)
+    @eval zero(a::$T) = $T(zero.(a.coeffs))
+    @eval function one(a::$T)
+        b = zero(a)
+        b[0] = one(b[0])
+        return b
+    end
 end
 
 function zero(a::HomogeneousPolynomial{T}) where {T<:Number}
     v = zero.(a.coeffs)
     return HomogeneousPolynomial(v, a.order)
 end
 
-function zeros(::HomogeneousPolynomial{T}, order::Int) where {T<:Number}
-    order == 0 && return [HomogeneousPolynomial([zero(T)], 0)]
+function zeros(a::HomogeneousPolynomial{T}, order::Int) where {T<:Number}
+    order == 0 && return [HomogeneousPolynomial([zero(a[1])], 0)]
     v = Array{HomogeneousPolynomial{T}}(undef, order+1)
     @simd for ord in eachindex(v)
-        @inbounds v[ord] = HomogeneousPolynomial([zero(T)], ord-1)
+        @inbounds v[ord] = HomogeneousPolynomial([zero(a[1])], ord-1)
```

Comment:
In this case you can avoid creating  the vector `[zero(a[1])]`

Reply:
Changed, thanks!